apps/wolfsshd/wolfsshd.c

```diff
@@ -1771,81 +1829,59 @@ static void* HandleConnection(void* arg)
-            /* check for any forced command set for the user */
-            switch (wolfSSH_GetSessionType(ssh)) {
-                case WOLFSSH_SESSION_SHELL:
-                #ifdef WOLFSSH_SHELL
-                    if (ret == WS_SUCCESS) {
-                        wolfSSH_Log(WS_LOG_INFO, "[SSHD] Entering new shell");
-                        SHELL_Subsystem(conn, ssh, pPasswd, usrConf, NULL);
-                    }
-                #else
-                    wolfSSH_Log(WS_LOG_ERROR,
-                        "[SSHD] Shell support is disabled");
-                    ret = WS_NOT_COMPILED;
-                #endif
-                    break;
-
-                case WOLFSSH_SESSION_SUBSYSTEM:
-                    /* test for known subsystems */
-                    switch (ret) {
-                        case WS_SFTP_COMPLETE:
-                        #ifdef WOLFSSH_SFTP
-                            ret = SFTP_Subsystem(conn, ssh, pPasswd, usrConf);
-                        #else
-                            err_sys("SFTP not compiled in. Please use "
-                                    "--enable-sftp");
-                        #endif
-                            break;
-
-                        case WS_SCP_INIT:
-                        #ifdef WOLFSSH_SCP
-                            ret = SCP_Subsystem(conn, ssh, pPasswd, usrConf);
-                        #else
-                            err_sys("SCP not compiled in. Please use "
-                                    "--enable-scp");
-                        #endif
-                            break;
-
-                        default:
-                            wolfSSH_Log(WS_LOG_ERROR,
-                                "[SSHD] Unknown or build not supporting sub"
-                                "system found [%s]",
-                                wolfSSH_GetSessionCommand(ssh));
-                            ret = WS_NOT_COMPILED;
-                    }
-                    break;
-
-                case WOLFSSH_SESSION_UNKNOWN:
-                case WOLFSSH_SESSION_EXEC:
-                #if defined(WOLFSSH_SHELL)
-                    if (ret == WS_SUCCESS) {
-                        wolfSSH_Log(WS_LOG_INFO,
-                            "[SSHD] Entering exec session [%s]",
-                                wolfSSH_GetSessionCommand(ssh));
-                        SHELL_Subsystem(conn, ssh, pPasswd, usrConf,
-                                wolfSSH_GetSessionCommand(ssh));
-                    }
-                #endif /* WOLFSSH_SHELL */
-
-                    /* SCP can be an exec type */
-                    if (ret == WS_SCP_INIT) {
-                    #ifdef WOLFSSH_SCP
-                        ret = SCP_Subsystem(conn, ssh, pPasswd, usrConf);
-                    #else
-                        err_sys("SCP not compiled in. Please use "
-                                "--enable-scp");
-                    #endif
-                    }
-                    break;
+        do {
```

Comment:
needs sanity check on ret value to account for error in getting user info, before entering do while loop

Reply:
will this handle a peer just closing down the socket and disconnecting?

Reply:
If the client application authenticates the user then closes the socket, we should just close down. The client should also be able to send a MSG_DISCONNECT and the server should close out the connection.

Reply:
And you are right about the check on the return value.

Reply:
I probably should have been more clear about the peer just closing down. Disconnected as in the Ethernet wire was pulled or internet connection shut off. I think this do while loop would continue spinning on in that case since the channel is never opened?